crates/store/src/state.rs

```diff
@@ -451,6 +450,34 @@ impl State {
         Ok((state_sync, delta))
     }
 
+    /// Loads data to synchronize a client's notes.
+    ///
+    /// The client's request contains a list of tag prefixes, this method will return the first
+    /// block with a matching tag, or the chain tip. All the other values are filter based on this
+    /// block range.
+    ///
+    /// # Arguments
+    ///
+    /// - `block_num`: The last block *known* by the client, updates start from the next block.
+    /// - `note_tags`: The tags the client is interested in, resulting notes are restricted to
+    ///    the first block containing a matching note.
+    #[instrument(target = "miden-store", skip_all, ret(level = "debug"), err)]
+    pub async fn sync_notes(
+        &self,
+        block_num: BlockNumber,
+        note_tag_prefixes: Vec<u32>,
+    ) -> Result<(NoteSyncUpdate, MmrProof), NoteSyncError> {
+        let inner = self.inner.read().await;
+
+        let note_sync = self.db.get_note_sync(block_num, note_tag_prefixes).await?;
+
+        let mmr_proof = inner
+            .chain_mmr
+            .open(note_sync.block_header.block_num() as usize, inner.chain_mmr.forest())?;
```

Comment:
I just stole this from another method - is this correct?

Reply:
Yes, this is correct.